src/__tests__/bim.js

```diff
@@ -0,0 +1,245 @@
+                video: item.Video,
+                tag: item.Tag,
+                release: item.Release,
+                new: item.New,
+                order: item.Order,
+                sotd: item.SOTD,
+            }
+    );
+    return filterExcelData(reconData, ["Release 1", "Release 2"]);
+};
+
+const filterExcelData = (excelData, releases) => {
+    return excelData
+        .filter((group) => group !== false) // filter out those without any value
+        .filter((group) =>
+            Array.isArray(releases)
+                ? releases.includes(group.release)
+                : group.release === releases
+        ) // filter out those that are not in 'release'
+        .sort(
+            (a, b) => a.kumpulanKategori.localeCompare(b.kumpulanKategori) // sort the entries alphabetically based on the Kategori
+        );
+};
+
+// For Sheet[1]
+const restructureJSONGroup = (data) => {
+    const reconData = data.map(
+        (item) =>
+            item.KumpulanKategori !== undefined &&
+            item.GroupCategory !== undefined && {
+                kumpulanKategori: item.KumpulanKategori.trim(),
+                groupCategory: item.GroupCategory.trim(),
+                remark: item.Remark,
+            }
+    );
+    return reconData.filter((group) => group !== false);
+};
+
+// Read xlsx in public
+const readExcelTest = async (sheet) => {
+    const file = "public/assets/BIM.xlsx";
+    const promise = new Promise((resolve, reject) => {
+        const wb = XLSX.readFile(file);
+        const wsVocabs = wb.Sheets[wb.SheetNames[sheet]];
+        const data = XLSX.utils.sheet_to_json(wsVocabs);
+        const reconData =
+            sheet === 0 ? restructureJSON(data) : restructureJSONGroup(data);
+        resolve(reconData);
+    });
+    return promise;
+};
+
+beforeEach(() => {
+    readExcelTest(0).then((value) => {
+        Dispatcher.dispatch({
+            actionType: "STORE_EXCEL",
+            payload: value,
+        });
+    });
+    readExcelTest(1).then((value) => {
+        Dispatcher.dispatch({
+            actionType: "STORE_EXCEL_GROUP",
+            payload: value,
+        });
+    });
+});
+
+// Test for BIM.xlsx file existence
+test("Test 1: BIM.xlsx should exist at public/assets/BIM.xlsx ", () => {
+    const dirBIM = "public/assets/BIM.xlsx";
+    expect.extend({
+        fileExist(dir) {
+            try {
+                const pass = existsSync(dir);
+
+                if (pass) {
+                    return {
+                        message: () => `BIM.xlsx is in the correct directory.`,
+                        pass: true,
+                    };
+                } else {
+                    return {
+                        message: () =>
+                            `BIM.xlsx does not exist. Please add the BIM into the correct directory.`,
+                        pass: false,
+                    };
+                }
+            } catch (error) {
+                console.error(error);
```

Comment:
are these console.error necessary? If there is an error, it will be thrown either ways.